src/SignalR/perf/benchmarkapps/BenchmarkServer/Hubs/EchoHub.cs

```diff
@@ -30,6 +35,29 @@ public async Task Broadcast(int duration)
             Console.WriteLine("Broadcast exited: Sent {0} messages", sent);
         }
 
+        public static string Status => $"{_connectionCount} current, {_peakConnectionCount} peak.";
+
+        public void LogConnections(string label) {
+            if (_connectionCount < 1000 || _connectionCount % 100 == 0)
+            {
+                var timeSinceServerStart = DateTime.Now.Subtract(_serverStart).ToString(@"hh\:mm\:ss");
+                Console.WriteLine($"[{timeSinceServerStart}] {label}: {Status}");
+            }
+        }
+
+        public override Task OnConnectedAsync() {
+            _connectionCount++;
+            _peakConnectionCount = Math.Max(_connectionCount, _peakConnectionCount);
+            LogConnections("Connected");
+            return Task.CompletedTask;
+        }
+
+        public override Task OnDisconnectedAsync(Exception exception) {
+            _connectionCount--;
```

Comment:
```suggestion
            Interlocked.Decrement(ref _connectionCount);
```

https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.decrement?view=netframework-4.7.2